src/test/java/seedu/address/model/person/MajorTest.java

```diff
@@ -0,0 +1,77 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class MajorTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Major(null));
+    }
+
+    @Test
+    public void constructor_invalidMajor_throwsIllegalArgumentException() {
+        String invalidMajor = "invalid";
+        assertThrows(IllegalArgumentException.class, () -> new Major(invalidMajor));
+    }
+
+    @Test
+    public void isValidMajor() {
+        // null major
+        assertThrows(NullPointerException.class, () -> Major.isValidMajor(null));
+
+        // invalid majors
```

Comment:
I think you should include test for a single or multiple whitespace, since it would likely be a very common occurence when using this app.